wicket-core/src/main/java/org/apache/wicket/markup/html/image/ExternalImage.java

```diff
@@ -0,0 +1,298 @@
+	{
+		if (srcModel != null)
+		{
+			tag.put("src", srcModel.getObject().toString());
+		}
+	}
+
+	/**
+	 * Builds the srcset attribute if multiple models are found as varargs
+	 *
+	 * @param tag
+	 *            the component tag
+	 * @param srcSetModels
+	 *            the models containing the src set URLs
+	 */
+	protected void buildSrcSetAttribute(final ComponentTag tag,
+		IModel<List<Serializable>> srcSetModels)
+	{
+		int srcSetPosition = 0;
+		List<Serializable> srcSetItems = srcSetModels.getObject();
+		if (srcSetItems != null)
+		{
+			for (Serializable srcSetModel : srcSetItems)
+			{
+				String srcset = tag.getAttribute("srcset");
+				String xValue = "";
+
+				// If there are xValues set process them in the applied order to the srcset
+				// attribute.
+				if (xValues != null)
+				{
+					xValue = xValues.size() > srcSetPosition && xValues.get(srcSetPosition) != null
+						? " " + xValues.get(srcSetPosition)
+						: "";
+				}
+				tag.put("srcset", (srcset != null ? srcset + ", " : "") + srcSetModel + xValue);
+				srcSetPosition++;
+			}
+		}
+	}
+
+	/**
+	 * builds the sizes attribute of the img tag
+	 *
+	 * @param tag
+	 *            the component tag
+	 */
+	protected void buildSizesAttribute(final ComponentTag tag)
+	{
+		// if no sizes have been set then don't build the attribute
+		if (sizes == null)
+		{
+			return;
+		}
+		String sizes = "";
+		for (String size : this.sizes)
+		{
+			sizes += size + ",";
+		}
+		int lastIndexOf = sizes.lastIndexOf(",");
+		if (lastIndexOf != -1)
+		{
+			sizes = sizes.substring(0, lastIndexOf);
+		}
+		if (!"".equals(sizes))
+		{
+			tag.put("sizes", sizes);
+		}
+	}
+
+	/**
+	 * @param values
+	 *            the x values to be used in the srcset
+	 */
+	public void setXValues(String... values)
+	{
+		if (xValues == null)
+		{
+			xValues = new ArrayList<>();
+		}
+		else
+		{
+			xValues.clear();
+		}
+		xValues.addAll(Arrays.asList(values));
+	}
+
+	/**
+	 * @param sizes
+	 *            the sizes to be used in the size
+	 */
+	public void setSizes(String... sizes)
+	{
+		if (this.sizes == null)
+		{
+			this.sizes = new ArrayList<>();
+		}
+		else
+		{
+			this.sizes.clear();
+		}
+		this.sizes.addAll(Arrays.asList(sizes));
+	}
+
+	/**
+	 * Gets the cross origin settings
+	 * 
+	 * @see {@link org.apache.wicket.markup.html.image.Image#setCrossOrigin(Cors)}
+	 *
+	 * @return the cross origins settings
+	 */
+	public Cors getCrossOrigin()
+	{
+		return crossOrigin;
+	}
+
+	/**
+	 * Sets the cross origin settings
+	 * 
+	 * @see {@link org.apache.wicket.markup.html.image.Image#setCrossOrigin(Cors)}
+	 * @param crossOrigin
+	 *            the cross origins settings to set
+	 */
+	public void setCrossOrigin(Cors crossOrigin)
+	{
+		this.crossOrigin = crossOrigin;
+	}
+
+	/**
+	 * Gets a list of models containing the src set values
+	 * 
+	 * @return a list of models containing the src set values
+	 */
+	public IModel<List<Serializable>> getSrcSet()
```

Comment:
IMO: either it is named `#getSrcSetModel` and it returns the model, or it is named `#getSrcSet` and it returns the underlying model object :). The first one sound more logical...